File: backend-api/src/main/java/vn/aptech/backendapi/entities/Working.java
package vn.aptech.backendapi.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="workings")
public class Working extends  BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String company;
    private String address;
    @Column(name="start_work")
    private LocalDate startWork;

    @Column(name="end_work")
    private LocalDate endWork;

    private boolean status;

    @ManyToOne
    @JoinColumn(name = "doctor_id", referencedColumnName = "id")
    private Doctor doctor;
}

File: mobile/lib/models/working.dart
class Working{
  late final int id;
  late final String company;
  late final String address;
  late final String startWork;
  late final String endWork;
  late final bool status;
  late final int doctorId;
  Working({
    required this.id,
    required this.company,
    required this.address,
    required this.startWork,
    required this.endWork,
    required this.status,
    required this.doctorId,
  }) ;
  factory Working.fromJson(Map<String, dynamic> json) {
    return Working(
      id: json['id'],
      company: json['company'] ?? '',
      address: json['address'] ?? '',
      startWork: json['startWork'] ?? '',
      endWork: json['endWork'] ?? '',
      status: json['status'] as bool,
      doctorId: json['doctorId'] as int,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'company': company,
      'address': address,
     'startWork': startWork,
      'endWork': endWork,
     'status': status,
      'doctorId': doctorId,
    };
  }


}